### ISSUE TITLE
Is MacGap2 dead?

### ISSUE THREAD

**Issue by behrangsa** [MacGapProject/MacGap2#64]
Last commit is from 5 months ago. Is MacGap dead? Is it renamed to something else?

Is there a thorough documentation available for it? Any screencasts? Books?

Thanks,
Behrang

**Comment by jeff-h** [comment]
Hi — it's not dead, more a case of "it works well enough for 90% that we don't get pushed hard to implement something new".

Also, since it's aimed at Javascript developers (who usually don't also write Objective-C), there are very few users able to commit new features. Those that *can* commit new features typically aren't that interested in what MacGap does, since they could code their own native solution in most use-cases.

It's a case where the typical open source "scratch-your-own-itch" motivation doesn't quite fit the equation, unfortunately. There are a few recent issues which have remained without a good solution for exactly this reason e.g. https://github.com/MacGapProject/MacGap2/issues/62

To my knowledge there is no better solution for close-to-native web apps on OS X.

**Comment by behrangsa** [comment]
HI @jeff-h,

Thanks for your reply. Liveness of the project aside, are there any good tutorials, screencasts, articles, etc. for getting started with MacGap and building an app? Does MacGap have an API docs similar to Java, Cocoa, or, say, Angular JS? For People who are familiar with PhoneGap, it might be easy to get started and write MacGap apps, but I haven't used PhoneGap before so I am looking for a _MacGap: The Beginners Guide_ kind of introduction/book/article.

Also while the functionality of MacGap might be complete enough it is still lackadaisical on the documentation side of things. And no progress being made in this area, makes MacGap more or less an unmaintained project.

**Comment by puccaso** [comment]
Hey there guys, 
Love the app. 
I can completely under the 90% working user case, but with recent changes to Mojave and xcode, plus safari12's new security requirements, I wonder if a tiny update might be required? or at least that what i'm requesting. 

Xcode gives a plethora of things to be changed, MenuItems dont conform, loads of stubs etc - 
to be honest, i dont even know what a stub is - but xcode wants to put it in the project, and things break.

love the code!

**Comment by jeff-h** [comment]
Strange — it compiles just fine for me (Mojave 10.14.1, Xcode 10.1). It did throw a bunch of warnings but that's not too unexpected. It did suggest adding one stub but gave me nothing about MenuItems. Are you on an older version of Xcode perhaps?

Regardless, I just updated the code to remove most of the warnings.

In terms of the future of MacGap, would you mind sharing a few thoughts on why MacGap appeals? For me it was primarily to avoid the enormous & unnecessary weight of Electron, but that feels like a lost battle in 2018.

**Comment by tcaer** [comment]
I personally agree that the weight of Electron is a big problem for me. It is something I can feel when running them on my machine.

**Comment by blakewatson** [comment]
@jeff-h Yeah my use case doesn't require cross-platform capability (in which case I should probably be using native code instead of HTML/CSS/JS). I just want to make desktop apps on the Mac with web tech—which, admittedly, isn’t a compelling reason for you to continue work on it. :-)

**Comment by blakewatson** [comment]
I still say a lightweight alternative to Electron would be great though. For developers and users.

**Comment by jeff-h** [comment]
Yeah — I definitely believe in MacGap and its place in the world. I just wish someone else would write it lol I'd love to dive into it, but I've learnt enough to know that once the fun fades and it becomes hard work it's going to get "procrastinated"...
